From: Tommi <tommi@webrtc.org>
Date: Mon, 26 Jun 2023 21:11:35 +0200
Subject: (cherry-pick-branch-heads/5790) [M115] Revert mid check in
 SdpOfferAnswerHandler::CreateDataChannel.

This check was added here:
  https://webrtc-review.googlesource.com/c/src/+/300544

When createOffer is used before createAnswer, this check would cause
SetupDataChannelTransport_n to not be called for the remote channel.

(cherry picked from commit 299cdc9057fb8925f1692ac8c7573f51b7872df6)

Bug: webrtc:15258
Change-Id: Ifdab35d1b0260ff03fef4beff13acf8090d59d8f
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/310460
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#40357}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/313620
Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
Cr-Commit-Position: refs/branch-heads/5790@{#8}
Cr-Branched-From: 2eacbbc03a4a41ea658661225eb1c8fc07884c33-refs/heads/main@{#40122}
---
 pc/sdp_offer_answer.cc | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/pc/sdp_offer_answer.cc b/pc/sdp_offer_answer.cc
index 6596d82167..4874444eae 100644
--- a/pc/sdp_offer_answer.cc
+++ b/pc/sdp_offer_answer.cc
@@ -5145,11 +5145,7 @@ RTCError SdpOfferAnswerHandler::CreateChannels(const SessionDescription& desc) {
 
 bool SdpOfferAnswerHandler::CreateDataChannel(const std::string& mid) {
   RTC_DCHECK_RUN_ON(signaling_thread());
-  if (pc_->sctp_mid().has_value()) {
-    RTC_DCHECK_EQ(mid, *pc_->sctp_mid());
-    return true;  // data channel already created.
-  }
-
+  RTC_DCHECK(!pc_->sctp_mid().has_value() || mid == pc_->sctp_mid().value());
   RTC_LOG(LS_INFO) << "Creating data channel, mid=" << mid;
 
   absl::optional<std::string> transport_name =
-- 
2.39.3 (Apple Git-145)

